src/request.ts

```diff
@@ -121,13 +127,25 @@ export class HonoRequest<P extends string = '/', I extends Input['out'] = {}> {
   }
 
   async parseBody<T extends BodyData = BodyData>(): Promise<T> {
-    return await parseBody(this)
+    if (this.bodyCache.parsedBody) return this.bodyCache.parsedBody as T
+    const parsedBody = await parseBody<T>(this)
+    this.bodyCache.parsedBody = parsedBody
+    return parsedBody
   }
 
-  private cachedBody = (key: keyof BodyCache) => {
+  private cachedBody = (key: keyof Body) => {
     const { bodyCache, raw } = this
     const cachedBody = bodyCache[key]
     if (cachedBody) return cachedBody
+    /**
+     * If an arrayBuffer cache is exist,
+     * use it for creating a text, json, and others.
+     */
+    if (bodyCache.arrayBuffer) {
+      return (async () => {
+        return await new Response(bodyCache.arrayBuffer)[key]()
+      })()
+    }
```

Comment:
we can just return `new Response(...` since it return a promise anyway

or we can just change the cachedBody function into an `async function` and whenever a function need to call it, we just need to await it